pkg/postage/listener/listener.go

```diff
@@ -177,13 +177,24 @@ func (l *listener) processEvent(e types.Log, updater postage.EventUpdater) error
 	}
 }
 
-func (l *listener) Listen(from uint64, updater postage.EventUpdater) <-chan struct{} {
+func (l *listener) Listen(from uint64, updater postage.EventUpdater, initState *postage.BatchSnapshot) <-chan struct{} {
 	ctx, cancel := context.WithCancel(context.Background())
 	go func() {
 		<-l.quit
 		cancel()
 	}()
 
+	if initState != nil {
```

Comment:
this section is not correct btw, not is it unit tested. when processing events you need to update the block height after every event. please see how the individual events are being processed normally by the listener